test/cql-pytest/test_service_levels.py

```diff
@@ -27,6 +28,42 @@ def new_service_level(cql):
         cql.execute(f"DETACH SERVICE LEVEL FROM {cql.cluster.auth_provider.username}")
         cql.execute(f"DROP SERVICE LEVEL IF EXISTS {sl}")
 
+@contextmanager
+def new_service_level_with_params(cql, timeout=None, workload_type=None):
+    params = ""
+    if timeout != None and workload_type != None:
+        params = f"WITH timeout = {timeout} AND workload_type = '{workload_type}'"
+    elif timeout != None:
+        params = f"WITH timeout = {timeout}"
+    elif workload_type != None:
+        params = f"WITH workload_type = '{workload_type}'"
+
+    try:
+        sl = f"sl_{unique_name()}"
+        cql.execute(f"CREATE SERVICE LEVEL {sl} {params}")
+        yield sl
+    finally:
+        cql.execute(f"DROP SERVICE LEVEL IF EXISTS {sl}")
+
+# Some of the service levels operations depends on controller's update
+# which currently are done in 10s intervals.
```

Comment:
This is so sad for tests. Right now the entire cql-pytest test suite, 1333 tests, takes less than 150 seconds. a single test taking 10 seconds is really bad. This number should have been configurable in the Scylla source code, and then the test could reduce it :-( But maybe we can leave with this 10s thing now...

By the way, I thought this test was enterprise-only, does the whole thing work in the open-source repositor (to which you sent this pull request)?

Reply:
Service levels are present in open-source. They are extended in enterprise to include shares and support workload prioritization.
So I think you are referring to https://github.com/scylladb/scylla-enterprise/pull/3242, which also had the infamous sleeps in tests.

I tried to reduce it once again but I think right now it's not possible to write test that doesn't depend on the 10s interval of updating the service level controller. 
The good thing is there is a plan to migrate service levels to raft and it will remove the method of updating the controller in intervals.